frontend/src/component/feature/FeatureStrategy/NewFeatureStrategyCreate/NewFeatureStrategyCreate.test.tsx

```diff
@@ -1,20 +1,322 @@
 import { formatAddStrategyApiCode } from 'component/feature/FeatureStrategy/FeatureStrategyCreate/FeatureStrategyCreate';
+import { screen, waitFor, fireEvent } from '@testing-library/react';
+import { render } from 'utils/testRenderer';
+import { Route, Routes } from 'react-router-dom';
 
-test('formatAddStrategyApiCode', () => {
-    expect(
-        formatAddStrategyApiCode(
-            'projectId',
-            'featureId',
-            'environmentId',
-            { id: 'strategyId' },
-            'unleashUrl',
-        ),
-    ).toMatchInlineSnapshot(`
+import {
+    CREATE_FEATURE_STRATEGY,
+    UPDATE_FEATURE_ENVIRONMENT_VARIANTS,
+    UPDATE_FEATURE_STRATEGY,
+} from 'component/providers/AccessProvider/permissions';
+import { NewFeatureStrategyCreate } from './NewFeatureStrategyCreate';
+import { testServerRoute, testServerSetup } from 'utils/testServer';
+
+const server = testServerSetup();
+
+const featureName = 'my-new-feature';
+
+const setupProjectEndpoint = () => {
+    testServerRoute(server, '/api/admin/projects/default', {
```

Comment:
I'd remove all the fields in all data setups that are not relevant to the test and only leave required fields. If some string is required to work but not checked for value I'd call it 'irrelevant'